tests/test_fastnn/test_attention_core.py

```diff
@@ -0,0 +1,64 @@
+import math
+
+import torch
+from einops import rearrange
+
+TEST_TRITON = True
+try:
+    from fastfold.model.fastnn.kernel import fused_attention_core
+except:
+    print("Skip triton attention test!")
+    TEST_TRITON = False
+
+
+def torch_core_attention(q, k, v, mask, bias):
+
+    scaling = 1. / math.sqrt(q.size(-1))
+    q = q * scaling
+
+    logits = torch.matmul(q.float(), k.float().transpose(-1, -2))
+    logits += bias.float()
+    logits += (1e20 * (mask - 1))[..., :, None, None, :]
+
+    weights = torch.nn.functional.softmax(logits.float(), -1).to(dtype=q.dtype)
+
+    weighted_avg = torch.matmul(weights, v)
+
+    weighted_avg = rearrange(weighted_avg, 'b1 b2 h n d -> b1 b2 n (h d)')
+
+    return weighted_avg
+
+
```

Comment:
maybe we can use @pytest.mark.skipif(TEST_TRITON == False, reason="triton is not available"). So it will have a S mark in pytest to annotate skip.